From 1cf81a056ca4fcd6a97265deaff9d525f1922673 Mon Sep 17 00:00:00 2001
From: bengris32 <bengris32@protonmail.ch>
Date: Thu, 6 Mar 2025 19:27:12 +0530
Subject: [PATCH 5/8] ImsService: Export receivers that aren't exclusively
 registered for system broadcasts

Signed-off-by: bengris32 <bengris32@protonmail.ch>
Signed-off-by: Sarthak Roy <sarthakroy2002@gmail.com>
Change-Id: Ib6d435028db0d72634056db64aafdc8a2696c182

SX3 port: Rebased on the stock SX3 APK; keep protected-only ImsApp and ImsLocationHandler receivers unchanged.
---
diff --git a/smali/com/mediatek/ims/ImsConfigManager.smali b/smali/com/mediatek/ims/ImsConfigManager.smali
--- a/smali/com/mediatek/ims/ImsConfigManager.smali
+++ b/smali/com/mediatek/ims/ImsConfigManager.smali
@@ -115,7 +115,7 @@
 .end method
 
 .method public constructor <init>(Landroid/content/Context;[Lcom/mediatek/ims/ril/ImsCommandsInterface;)V
-    .locals 2
+    .locals 3
     .param p1, "context"    # Landroid/content/Context;
     .param p2, "imsRILAdapters"    # [Lcom/mediatek/ims/ril/ImsCommandsInterface;
 
@@ -192,7 +192,9 @@
     .line 149
     iget-object v1, p0, Lcom/mediatek/ims/ImsConfigManager;->mBroadcastReceiver:Landroid/content/BroadcastReceiver;
 
-    invoke-virtual {p1, v1, v0}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
+    const/4 v2, 0x2
+
+    invoke-virtual {p1, v1, v0, v2}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;I)Landroid/content/Intent;
 
     .line 151
     return-void
diff --git a/smali/com/mediatek/ims/ImsService$MyHandler.smali b/smali/com/mediatek/ims/ImsService$MyHandler.smali
--- a/smali/com/mediatek/ims/ImsService$MyHandler.smali
+++ b/smali/com/mediatek/ims/ImsService$MyHandler.smali
@@ -3255,7 +3255,9 @@
 
     move-result-object v7
 
-    invoke-virtual {v6, v7, v5}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
+    const/4 v8, 0x2
+
+    invoke-virtual {v6, v7, v5, v8}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;I)Landroid/content/Intent;
 
     .line 2737
     .end local v5    # "filter":Landroid/content/IntentFilter;
diff --git a/smali/com/mediatek/ims/MtkSuppServExt.smali b/smali/com/mediatek/ims/MtkSuppServExt.smali
--- a/smali/com/mediatek/ims/MtkSuppServExt.smali
+++ b/smali/com/mediatek/ims/MtkSuppServExt.smali
@@ -1547,7 +1547,7 @@
 .end method
 
 .method private registerBroadcastReceiver()V
-    .locals 3
+    .locals 4
 
     .line 465
     new-instance v0, Landroid/content/IntentFilter;
@@ -1580,7 +1580,9 @@
 
     iget-object v2, p0, Lcom/mediatek/ims/MtkSuppServExt;->mBroadcastReceiver:Landroid/content/BroadcastReceiver;
 
-    invoke-virtual {v1, v2, v0}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
+    const/4 v3, 0x2
+
+    invoke-virtual {v1, v2, v0, v3}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;I)Landroid/content/Intent;
 
     .line 472
     iget-object v1, p0, Lcom/mediatek/ims/MtkSuppServExt;->mContext:Landroid/content/Context;
diff --git a/smali/com/mediatek/ims/config/internal/ImsConfigController.smali b/smali/com/mediatek/ims/config/internal/ImsConfigController.smali
--- a/smali/com/mediatek/ims/config/internal/ImsConfigController.smali
+++ b/smali/com/mediatek/ims/config/internal/ImsConfigController.smali
@@ -267,7 +267,7 @@
 .end method
 
 .method public constructor <init>(Landroid/content/Context;ILcom/mediatek/ims/ril/ImsCommandsInterface;)V
-    .locals 7
+    .locals 8
     .param p1, "context"    # Landroid/content/Context;
     .param p2, "phoneId"    # I
     .param p3, "imsRilAdapter"    # Lcom/mediatek/ims/ril/ImsCommandsInterface;
@@ -514,7 +514,9 @@
 
     iget-object v5, p0, Lcom/mediatek/ims/config/internal/ImsConfigController;->mReceiver:Landroid/content/BroadcastReceiver;
 
-    invoke-virtual {v4, v5, v3}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
+    const/4 v7, 0x2
+
+    invoke-virtual {v4, v5, v3, v7}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;I)Landroid/content/Intent;
 
     .line 153
     iget-object v4, p0, Lcom/mediatek/ims/config/internal/ImsConfigController;->mRilAdapter:Lcom/mediatek/ims/ril/ImsCommandsInterface;
diff --git a/smali/com/mediatek/ims/internal/ImsVTProviderUtil.smali b/smali/com/mediatek/ims/internal/ImsVTProviderUtil.smali
--- a/smali/com/mediatek/ims/internal/ImsVTProviderUtil.smali
+++ b/smali/com/mediatek/ims/internal/ImsVTProviderUtil.smali
@@ -1818,7 +1818,7 @@
 .end method
 
 .method public setContextAndInitRefVTPInternal(Landroid/content/Context;)V
-    .locals 4
+    .locals 5
     .param p1, "context"    # Landroid/content/Context;
 
     .line 841
@@ -1901,7 +1901,9 @@
 
     iget-object v3, p0, Lcom/mediatek/ims/internal/ImsVTProviderUtil;->mFeatureValueReceiver:Lcom/mediatek/ims/internal/ImsVTProviderUtil$FeatureValueReceiver;
 
-    invoke-virtual {v2, v3, v0}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
+    const/4 v4, 0x2
+
+    invoke-virtual {v2, v3, v0, v4}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;I)Landroid/content/Intent;
 
     .line 859
     .end local v0    # "filter":Landroid/content/IntentFilter;
diff --git a/smali/com/mediatek/ims/rcs/MtkRcsCapabilityExchangeImpl.smali b/smali/com/mediatek/ims/rcs/MtkRcsCapabilityExchangeImpl.smali
--- a/smali/com/mediatek/ims/rcs/MtkRcsCapabilityExchangeImpl.smali
+++ b/smali/com/mediatek/ims/rcs/MtkRcsCapabilityExchangeImpl.smali
@@ -220,7 +220,7 @@
 .end method
 
 .method public constructor <init>(ILandroid/content/Context;Landroid/telephony/ims/stub/CapabilityExchangeEventListener;)V
-    .locals 5
+    .locals 6
     .param p1, "slotId"    # I
     .param p2, "context"    # Landroid/content/Context;
     .param p3, "listener"    # Landroid/telephony/ims/stub/CapabilityExchangeEventListener;
@@ -384,7 +384,9 @@
 
     invoke-direct {v3, v4}, Landroid/content/IntentFilter;-><init>(Ljava/lang/String;)V
 
-    invoke-virtual {v1, v2, v3}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
+    const/4 v5, 0x2
+
+    invoke-virtual {v1, v2, v3, v5}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;I)Landroid/content/Intent;
 
     goto :goto_0
 
diff --git a/smali/com/mediatek/wfo/impl/MwiService.smali b/smali/com/mediatek/wfo/impl/MwiService.smali
--- a/smali/com/mediatek/wfo/impl/MwiService.smali
+++ b/smali/com/mediatek/wfo/impl/MwiService.smali
@@ -132,7 +132,7 @@
 .end method
 
 .method public constructor <init>(Landroid/content/Context;)V
-    .locals 8
+    .locals 9
     .param p1, "context"    # Landroid/content/Context;
 
     .line 90
@@ -353,7 +353,9 @@
     .line 123
     iget-object v3, p0, Lcom/mediatek/wfo/impl/MwiService;->mMultiSimConfigChangedReceiver:Landroid/content/BroadcastReceiver;
 
-    invoke-virtual {p1, v3, v2}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
+    const/4 v8, 0x2
+
+    invoke-virtual {p1, v3, v2, v8}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;I)Landroid/content/Intent;
 
     .line 125
     .end local v2    # "multiSimConfigChanged":Landroid/content/IntentFilter;
diff --git a/smali/com/mediatek/wfo/impl/WfcHandler.smali b/smali/com/mediatek/wfo/impl/WfcHandler.smali
--- a/smali/com/mediatek/wfo/impl/WfcHandler.smali
+++ b/smali/com/mediatek/wfo/impl/WfcHandler.smali
@@ -3270,7 +3270,7 @@
 .end method
 
 .method private registerForBroadcast()V
-    .locals 3
+    .locals 4
 
     .line 578
     new-instance v0, Landroid/content/IntentFilter;
@@ -3323,7 +3323,9 @@
 
     iget-object v2, p0, Lcom/mediatek/wfo/impl/WfcHandler;->mReceiver:Landroid/content/BroadcastReceiver;
 
-    invoke-virtual {v1, v2, v0}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
+    const/4 v3, 0x2
+
+    invoke-virtual {v1, v2, v0, v3}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;I)Landroid/content/Intent;
 
     .line 590
     return-void
diff --git a/smali/com/mediatek/wfo/impl/WifiPdnHandler.smali b/smali/com/mediatek/wfo/impl/WifiPdnHandler.smali
--- a/smali/com/mediatek/wfo/impl/WifiPdnHandler.smali
+++ b/smali/com/mediatek/wfo/impl/WifiPdnHandler.smali
@@ -3412,7 +3412,7 @@
 .end method
 
 .method private registerForBroadcast()V
-    .locals 3
+    .locals 4
 
     .line 765
     new-instance v0, Landroid/content/IntentFilter;
@@ -3450,7 +3450,9 @@
 
     iget-object v2, p0, Lcom/mediatek/wfo/impl/WifiPdnHandler;->mReceiver:Landroid/content/BroadcastReceiver;
 
-    invoke-virtual {v1, v2, v0}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
+    const/4 v3, 0x2
+
+    invoke-virtual {v1, v2, v0, v3}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;I)Landroid/content/Intent;
 
     .line 772
     return-void
